python/fusion_engine_client/analysis/analyzer.py

```diff
@@ -361,6 +367,69 @@ def _calc_stats(input):
 
         self._add_figure(name="time_scale", figure=figure, title="Time Scale")
 
+    def plot_latency(self):
+        if self.output_dir is None:
+            return
+
+        if self.host_time_mapper is None:
+            return
+
+        # Read the pose data to get P1 and GPS timestamps.
+        # The message_index will be used to map back to the host times the message data was recieved.
+        # This will then be compared to the GPS time, assuming the host time was synchronized to GPS time.
+        result = self.reader.read(message_types=[PoseMessage], return_message_index=True,
+                                  source_ids=self.default_source_id, **self.params)
+        pose_data = result[PoseMessage.MESSAGE_TYPE]
+
+        valid_idx = ~np.isnan(pose_data.gps_time)
+
+        if np.sum(valid_idx) == 0:
+            return
+
+        p1_time = pose_data.p1_time[valid_idx]
+        gps_time = pose_data.gps_time[valid_idx]
+        message_index = pose_data.message_index[valid_idx]
+
+        last_gps_time = gps_time[-1]
+
+        # Setup the figure.
+        p1_time_axis_str = 'Relative Time' if self.time_axis == 'relative' else 'P1 Time'
```

Comment:
Looks like this was taken from the time scale plot, which I think is a bit outdated. You can use `self.p1_time_label` instead, which should do this automatically (and includes the units already).